libs/src/services/discoveryProvider/DiscoveryProvider.ts

```diff
@@ -1166,6 +1172,30 @@ export class DiscoveryProvider {
     return parsedResponse.data
   }
 
+  /**
+   * Retrieves the user's replica se
+   * @param params.encodedUserId string of the encoded user id
+   * @param params.blocNumber optional integer pass to wait until the discovery node has indexed that block number
+   * @return profile info of user attached to JWT payload if the JWT is valid, else false
+   */
+  async getUserReplicaSet({
+    encodedUserId,
+    blockNumber
+  }: {
+    encodedUserId: string
+    blockNumber?: number
+  }): Promise<Object | null | undefined> {
+    const req = Requests.getUserReplicaSet(encodedUserId)
+
+    return await this._makeRequest<Object | null>(
+      req,
+      true,
+      0,
```

Comment:
why set `retry = true` with `attemptedRetries = 0`?

Reply:
These are the defaults, I added it because these are positional arguments and I modified the make request method to take another positional argument of blocknumber to use to check if the node has indexed the block. 
This is used in the get user replica set from discovery after a blocknumber and seemed like the least intrusive way to do it.

Reply:
oh i misread this as maxRetries, its attemptedRetries and is incremented, not decremented
weird function signature, but this is fine